api/types/maintenance.go

```diff
@@ -26,13 +26,17 @@ import (
 )
 
 const (
-	// UpgraderKindKuberController is a short name used to identify the kube-controller-based
+	// UpgraderKindKubeController is a short name used to identify the kube-controller-based
 	// external upgrader variant.
 	UpgraderKindKubeController = "kube"
 
 	// UpgraderKindSystemdUnit is a short name used to identify the systemd-unit-based
 	// external upgrader variant.
 	UpgraderKindSystemdUnit = "unit"
+
+	// UpgraderKindTeleportUpdate is a short name used to identify the teleport-update
+	// external upgrader variant.
+	UpgraderKindTeleportUpdate = "binary"
```

Comment:
Isn't "binary" too generic? Maybe "teleport-update"?

Reply:
We went through a few variations of this (`updater`, `teleport-update`, `binary`).

@hugoShaka and I liked `binary` best, because it clearly differentiates it from the deprecated updater that uses scripts (`unit`) and the Kubernetes-based updater that runs a controller. No strong opinion though. `teleport-update` makes the inventory column quite long.